dds/DCPS/RTPS/BaseMessageUtils.cpp

```diff
@@ -209,15 +209,15 @@ bool bitmapNonEmpty(const SequenceNumberSet& snSet)
     return false;
   }
 
-  const size_t last_index = num_ulongs - 1;
+  const CORBA::ULong last_index = num_ulongs - 1;
```

Comment:
Will this generate a new warning for 64-bit to 32-bit truncation?

Reply:
Right.  I'll revert that.

Reply:
Looking at other usages of SequenceNumberSet, such as in the RtpsCoreTypeSupportImpl RtpsCoreTypeSupportImpl::operator<< function, Corba::ULong is used instead of size_t.  Should I follow that example and change bitmapNonEmpty to use Corba::ULong for num_ulongs and in the for loop?  I think that might be the correct change.

Reply:
Yes, I think so.  The reasoning is that IDL sequences use 32-bit unsigned integers for their lengths and subscripting functions.  Especially the way these particular sequences are used, there is no chance that we'll have more than 4.3 billion elements.

Reply:
Ok.  I'll make num_ulongs and the for loop use CORBA::ULong.